File: sdk/typescript/src/internal/origin.test.ts
import { expect, test } from "bun:test"
import vectors from "../../../../internal/origin/testdata/origins.json"
import { canonicalSecretOrigin } from "./origin"
for (const vector of vectors) test(`Secret origin: ${vector.input}`, () => {
 if (vector.canonical) expect(canonicalSecretOrigin(vector.input)).toBe(vector.canonical)
 else expect(() => canonicalSecretOrigin(vector.input)).toThrow()
})

File: internal/origin/origin_test.go
package origin

import (
	"encoding/json"
	"os"
	"testing"
)

func TestCanonicalSharedVectors(t *testing.T) {
	data, err := os.ReadFile("testdata/origins.json")
	if err != nil {
		t.Fatal(err)
	}
	var vectors []struct{ Input, Canonical string }
	if err := json.Unmarshal(data, &vectors); err != nil {
		t.Fatal(err)
	}
	for _, v := range vectors {
		t.Run(v.Input, func(t *testing.T) {
			got, err := Canonical(v.Input)
			if v.Canonical == "" {
				if err == nil {
					t.Fatalf("accepted %q", got)
				}
			} else if err != nil || got != v.Canonical {
				t.Fatalf("got %q, %v", got, err)
			}
		})
	}
}
